src/radical/pilot/utils/session.py

```diff
@@ -114,32 +113,44 @@ def fetch_profiles (sid, dburl=None, src=None, tgt=None, access=None,
             PROFILES_TARBALL = '%s.prof.tgz' % pilot['uid']
             tarball_available = False
             try:
-                if  sandbox.is_file(PROFILES_TARBALL) and \
-                    sandbox.get_size(PROFILES_TARBALL):
-
+                if  sandbox.is_file(PROFILES_TARBALL) and sandbox.get_size(PROFILES_TARBALL):
                     log.info("profiles tarball exists")
-                    ftgt = rs.Url('%s/%s' % (tgt_url, PROFILES_TARBALL))
-
-                    if skip_existing and os.path.isfile(ftgt.path) \
-                            and os.stat(ftgt.path).st_size > 0:
-
-                        log.info("skip fetching of '%s/%s' to '%s'.",
-                                 sandbox_url, PROFILES_TARBALL, tgt_url)
-                        tarball_available = True
-                    else:
-
-                        log.info("fetch '%s%s' to '%s'.", sandbox_url,
-                                 PROFILES_TARBALL, tgt_url)
-
-                        prof_file = rs.fs.File("%s%s" % (sandbox_url,
-                                            PROFILES_TARBALL), session=session)
-                        prof_file.copy(ftgt, flags=rs.fs.CREATE_PARENTS)
-                        prof_file.close()
+                else:
+                    log.warn("profiles tarball doesnt exists! creating now.")
+                    # so lets create a tarball right now with SAGA JobService
+                    js_url = pilot['js_hop']
+                    log.debug('js  : %s', js_url)
+                    js = rs.job.Service(js_url, session=session)
+
+                    # find <pilot_sandbox> -name \*.prof > profiles.lst; tar jcf profiles.tbz -T profiles.lst
+                    cmd = "find %s -name \\*.prof  -exec tar -czf %s/%s {} +" % (sandbox.url.path, sandbox.url.path,
```

Comment:
I tried using profiles.lst method, but was facing absolute path problems then. Later figured out it was because of something else and profiles.lst path itself. 
I can change the logic to > profiles.lst path way. NP

Reply:
The command is likely going to fail (or to yield unexpected results such as incomplete tarballs) for large numbers of files: the total length of the command line `find` can build up this way is limited, usually to something like 256 characters. 

Also, you may want to use bzip2 compression (`tar cjf` in instead of `tar czf`) as that compresses much better than `zip` for our profiles.  Or at least I think it does, you may want to check maybe...

Reply:
1. Will change to profiles.lst way

2. I used `z` as gzip just because later in the code it was using  - 
```                    tarball = tarfile.open(ftgt.path, mode='r:gz')
```
which I thought as gzip. Also, the file name ending with `.tgz`.
Should I change to bzip2?

Reply:
What was the problem you faced with paths?  You could change the command line from `find $pilot_sandbox -name \*.prof ...` to `cd $pilot_sandbox; find . -name \*.prof` maybe?

Reply:
The path problems were my deficit understanding of these commands. I found find can run command via `-exec` option so used it, and by then the path things were fixed.
So i didn't changed to file way. 

Now, updated the command to use `profiles.lst` file and passing it to tar. 
Using `cd` before running everything, makes it clearer. 


Lastly, still using c**z**f with tarball.

Reply:
> Lastly, still using czf with tarball.

how did you decide - did you measure how both versions behave? :-)

Reply:
I have changed to bzip2 algo. I did not any measurement. 
As mentioned above, I decided based on gzip mode `r:gz` that was expected with in extraction and naming `tgz`.

I also realized that after switching to bzip2 the tarball size was reduced to 21K from 34K (in case of gzip).